tensorflow_addons/text/crf.py

```diff
@@ -0,0 +1,448 @@
+# Copyright 2019 The TensorFlow Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+import numpy as np
+import tensorflow as tf
+
+
+@tf.function
```

Comment:
Do we want to annotate these functions with `tf.function`?

`max_seq_len` usually varies a lot between batches which would cause many retraces. An input signature should at least be set, but even then there is a current TF issue which causes graphs to be rebuilt internally when the input shape changes, see https://github.com/tensorflow/tensorflow/issues/29075.

Reply:
I've removed the `tf.function` wrappers, and added a TODO.